docs-v2/pages/connect/components.mdx

```diff
@@ -0,0 +1,501 @@
+    "configurable_props": [
+      {
+        "name": "gitlab",
+        "type": "app",
+        "app": "gitlab"
+      },
+      {
+        "name": "projectId",
+        "type": "integer",
+        "label": "Project ID",
+        "description": "The project ID, as displayed in the main project page",
+        "remoteOptions": true
+      },
+      {
+        "name": "refName",
+        "type": "string",
+        "label": "Branch Name",
+        "description": "The name of the branch",
+        "remoteOptions": true,
+        "optional": true
+      },
+      {
+        "name": "max",
+        "type": "integer",
+        "label": "Max Results",
+        "description": "Max number of results to return. Default value is `100`",
+        "optional": true,
+        "default": 100
+      }
+    ]
+  }
+}
+```
+
+Using this information, you can now drive the configuration of the component for
+your end users, as described in the next section.
+
+### Configuration
+
+Component execution on behalf of your end users requires a few preliminary
+steps, focused on getting the right input parameters (a.k.a.
+[props](/workflows/using-props)) to the component.
+
+Configuring each prop for a component usually involves an API call to our
+backend to retrieve the possible values, unless the values that a prop can take
+are static/free-form. The endpoint is accessible at:
+
+```text
+POST /v1/connect/components/configure
+```
+
+The reason for this back-and-forth is because the options for a prop are usually
+linked to a specific user's account, and the Pipedream API is the one in charge
+of contacting the third-party service using the user's credentials, retrieving
+the necessary data, and sending these options back as a response. Please not
+that this logic is not implemented by the API itself, but by the component's
+code; in particular, the `options` function of the corresponding prop.
+
+The initial configuration call must contain your user's account ID for the
+target app, and the name of the prop you want to configure. Using the ["List
+Commits" component for
+Gitlab](https://github.com/PipedreamHQ/pipedream/blob/master/components/gitlab/actions/list-commits/list-commits.mjs#L4)
+as an example, the payload sent to the configuration API would look like this:
+
+```json
+{
+  "async_handle": "qxyTp7QcS2A6",
+  "external_user_id": "demo-34c13d13-a31e-4a3d-8b63-0ac954671095",
+  "id": "gitlab-list-commits",
+  "prop_name": "projectId",
+  "configured_props": {
+    "gitlab": {
+      "authProvisionId": "apn_oOhaBlD"
+    }
+  }
+}
+```
+
+The goal of this particular call is to retrieve the options for the `projectId`
+prop of the "List Commits" component for Gitlab, for the user with the ID
+`demo-34c13d13-a31e-4a3d-8b63-0ac954671095`. The `authProvisionId` is the ID of
+the account that the user has connected to the Gitlab app (see [this
+section](workflows#configure-accounts-to-use-your-end-users-auth) for more
+details on how to create these accounts).
```

Comment:
This should probably just be combined with the section above if you're documenting all params there.